addons/survey/controllers/main.py

```diff
@@ -376,6 +376,7 @@ def _prepare_question_html(self, survey_sudo, answer_sudo, **post):
         elif 'page' in survey_data:
             background_image_url = survey_data['page'].background_image_url
 
+        survey_data['survey'].session_question_start_time = datetime.now()
```

Comment:
Why this by the way ?

Reply:
because in some weird cases session_question_start_time was undefined or not recomputed (so the value of previous questions was used). So I decided that maybe having a session time calculated here all the time to make sure that we don't have errors because of some specific weird cases.

Reply:
As for adding an extra second or two on questions so that people can get "full scores", it may be even more confusing for the users to have a fixed amount "hidden seconds" that they are not aware of, especially when we don't know the difficulty of questions they want to put in the survey. user can decide that questions are super easy and people should solve them in 5-10 seconds, or have 1-5 minutes per question, and in both cases having 2 second correction would be useless, in first case because it'll defeat the purpose of checking who answers questions fastest, and in second case it won't make any difference at all. 

At some point maybe it would indeed be good to have an option for users to specify if they want users to get full marks if they answer in X seconds though, as I think some people would definitely like to have this option as long as they are aware that it exists.

Reply:
What is a bit strange is that you will never obtain maximum score (due to time reading the question, click on answer, ...). But indeed it would be confusing to have some magic that is never explained. Anyway it was just a thought when passing by ^^ .

Reply:
I'm choosing to keep the 2 seconds delay exactly for that (slightly fixed^^): reading the question and rewarding fully if you know immediately. This doesn't feel magical to me; the only missing thing would be a good-looking dynamic reverse progress bar showing the "points if validating now" so the user sees this 2sec "free" during the session.